File: 1704-special-positions-in-a-binary-matrix/special-positions-in-a-binary-matrix.py
class Solution:
    def numSpecial(self, mat: List[List[int]]) -> int:
        m, n = len(mat), len(mat[0])
        row_sum = [0] * m
        col_sum = [0] * n
        for r in range(m):
            for c in range(n):
                if mat[r][c] == 1:
                    row_sum[r] += 1
                    col_sum[c] += 1

        res = 0
        for r in range(m):
            for c in range(n):
                if mat[r][c] == 1 and row_sum[r] == 1 and col_sum[c] == 1:
                    res += 1

        return res  

File: 1704-special-positions-in-a-binary-matrix/special-positions-in-a-binary-matrix.java
class Solution {
    public int numSpecial(int[][] mat) {
        int m = mat.length;
        int n = mat[0].length;
        int[] rowOnes = new int[m];
        int[] colOnes = new int[n];

        for (int r = 0; r < m; r++) {
            for (int c = 0; c < n; c++) {
                if (mat[r][c] == 1) {
                    rowOnes[r]++;
                    colOnes[c]++;
                }
            }
        }

        int count = 0;
        for (int r = 0; r < m; r++) {
            for (int c = 0; c < n; c++) {
                if (mat[r][c] == 1 && rowOnes[r] == 1 && colOnes[c] == 1) {
                    count++;
                }
            }
        }
        return count;
    }
}